src/combinatorics/set_family/compression/uv.lean

```diff
@@ -52,6 +49,56 @@ branch `combinatorics`.
 compression, UV-compression, shadow
 -/
 
+section generalized_boolean_algebra
+variables {α : Type*} [generalized_boolean_algebra α] {x y z : α}
+
+lemma disjoint.le_sdiff_of_sup_le_left (hxz : disjoint x z) (h : z ⊔ x ≤ y) : x ≤ y \ z :=
+hxz.symm.sup_sdiff_cancel_left.ge.trans (sdiff_le_sdiff_right h)
+
+lemma inf_sdiff_left_comm : x \ z ⊓ y = (x ⊓ y) \ z :=
+by rw [@inf_comm _ _ x, inf_comm, inf_sdiff_assoc]
+
+end generalized_boolean_algebra
+
+namespace finset
+variables {α : Type*} [decidable_eq α] {s t u : finset α} {a : α}
+
+lemma erase_eq (s : finset α) (a : α) : s.erase a = s \ {a} := (sdiff_singleton_eq_erase _ _).symm
```

Comment:
This probably isn't a lemma we need, given it's the reverse of an existing one!

Reply:
This matches `finset.insert_eq` in functionality.

Reply:
#18729